### ISSUE TITLE
NPE in courier AI [1.21.1]

### ISSUE THREAD

**Issue by IchHabeHunger54** [ldtteam/minecolonies#10259]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.21

### MineColonies Version

1.1.702-1.21.1-snapshot

### Structurize Version

1.0.751-1.21.1-snapshot

### Related Mods and their Versions

NeoForge 21.1.54
Others should not be relevant, can provide if needed.

### Current Behavior

My courier ran into an AI issue. I scanned the logs myself for a stacktrace, but the only thing I found was the following:
```
[22Sep2024 21:20:25.967] [Server thread/WARN] [minecolonies/]: Statemachine for state DELIVERY threw an exception:
java.lang.NullPointerException: null
[22Sep2024 21:20:25.968] [Server thread/ERROR] [minecolonies/]: Pausing Citizen Kunneke Leymbecker (com.minecolonies.job.deliveryman) in colony:1 at BlockPos{x=1532, y=107, z=-847} for 6400 Seconds because of error:
```
The actual exception appears to be swallowed, which is always lovely to have.

### Expected Behavior

Regular courier behavior. It's not a big deal, they still mostly work like expected.

### Reproduction Steps

I honestly have no idea, it just happened to me today because I built my first courier today. Might be related to #10258?

### Logs

https://gist.githubusercontent.com/IchHabeHunger54/17b1509541c7c4b6592dfe0b7f3f50ba/raw/63dea101b6fa60825e405f85c7a34385fa5d810c/gistfile0.txt (WARNING: This is a big file because a dungeon mod keeps adding LOTS of debug info, relevant bit is near the end)

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Mikec78660** [comment]
This was fixed in 1.1.705-1.21.1 -snapshot. I would upgrade to the latest as they fixed a few more bugs as well.

**Comment by IchHabeHunger54** [comment]
Thank you for the information.
